include/variant_detection/variant_detection.hpp

```diff
@@ -1,39 +1,54 @@
 #pragma once
 
 #include <seqan3/std/filesystem>    // for filesystem
+#include <map>
 #include <vector>
 
 #include "iGenVar.hpp"              // for struct cmd_arguments
 #include "structures/junction.hpp"  // for class Junction
 
+/*! \brief Reads the header of the input file. Checks if input file is sorted and reads the reference sequence
+ *         dictionary.
```

Comment:
I like your solution of obtaining the ref_ids in the `read_header_information` function. However, you should document all return values and their meanings.

```suggestion
/*! \brief Reads the header of the input file. Checks if input file is sorted and reads the reference sequence
 *         dictionary. Stores the reference sequence lengths in parameter `reference_lengths` and returns the list
 *         of reference sequences.
```

Reply:
🙈